src/pages/workspace/WorkspaceNewRoomPage.js

```diff
@@ -76,19 +76,85 @@ class WorkspaceNewRoomPage extends React.Component {
         this.setState({workspaceOptions: _.map(workspaces, policy => ({label: policy.name, key: policy.id, value: policy.id}))});
     }
 
+    validateAndCreatePolicyRoom() {
+        if (!this.validate()) {
+            return;
+        }
+        Report.createPolicyRoom(
+            this.state.policyID,
+            this.state.roomName,
+            this.state.visibility,
+        );
+    }
+
     /**
-     * Called when a workspace is selected.
-     * @param {String} policyID
+     * @returns {Boolean}
      */
-    onWorkspaceSelect(policyID) {
-        this.setState({policyID});
+    validate() {
+        const errors = {};
+
+        const isExistingRoomName = _.some(
+            _.values(this.props.reports),
+            report => report && report.policyID === this.state.policyID
+            && report.reportName === this.state.roomName,
+        );
+
+        // We error if the user doesn't enter a room name or left blank
+        if (!this.state.roomName || this.state.roomName === CONST.POLICY.ROOM_PREFIX) {
+            errors.roomName = this.props.translate('newRoomPage.pleaseEnterRoomName');
+        }
+
+        // We error if the room name already exists. We don't care if it matches the original name provided in this
+        // component because then we are not changing the room's name.
+        if (isExistingRoomName) {
+            errors.roomName = this.props.translate('newRoomPage.roomAlreadyExistsError');
+        }
+
+        // Certain names are reserved for default rooms and should not be used for policy rooms.
+        if (_.contains(CONST.REPORT.RESERVED_ROOM_NAMES, this.state.roomName)) {
+            errors.roomName = this.props.translate('newRoomPage.roomNameReservedError');
+        }
+
+        if (!this.state.policyID) {
+            errors.policyID = this.props.translate('newRoomPage.pleaseSelectWorkspace');
+        }
+
+        this.setState({errors});
+        return _.isEmpty(errors);
```

Comment:
Please dry this as well.

Reply:
We can dry this up but code will become messy. I have defined why we cannot dry validate part [here](https://github.com/Expensify/App/pull/7098#discussion_r790137091).

Reply:
@parasharrajat please look at the comment and give me reply or suggestions on it.

Reply:
@parasharrajat what should I do for that one?